autogpt_platform/frontend/src/app/(platform)/build/stores/nodeStore.ts

```diff
@@ -44,10 +46,26 @@ export const useNodeStore = create<NodeStore>((set, get) => ({
     set((state) => ({
       nodeCounter: state.nodeCounter + 1,
     })),
-  onNodesChange: (changes) =>
+  onNodesChange: (changes) => {
+    const prevState = {
+      nodes: get().nodes,
+      connections: useEdgeStore.getState().connections,
+    };
+    const shouldTrack = changes.some(
+      (change) =>
+        change.type === "remove" ||
+        change.type === "add" ||
+        (change.type === "position" && change.dragging === false),
+    );
     set((state) => ({
       nodes: applyNodeChanges(changes, state.nodes),
-    })),
+    }));
+
+    if (shouldTrack) {
+      useHistoryStore.getState().pushState(prevState);
```

Comment:
This is called like a hook ( `use...` ), but it's really a regular function, so let's just call it `historyStore()` to prevent confusion 👍🏽